contents/performance/Performance.md

```diff
@@ -0,0 +1,102 @@
+
+Whenever data is requested, the computer will first look in the cache for the data. If it exists, this is known as a **cache hit**. If it does not exist, this is known as a **cache miss**. When a **cache miss** happens, a contiguous block of memory (not just the requested data) is retrieved and stored onto the cache. Because of this, we want to remember these rules to make cache friendly codes:
+
+* Temporal Locality
+* Spatial Locality
+* Row/Column Major Order
+
+#### Temporal Locality
+
+This rule states that recently used memory will likely be used in the near future. This means that making the scope of your variables smaller helps with execution times, as it will likely result in less cache misses.
+
+#### Spatial Locality
+
+This rule states that memory stored near each other will likely be used in the near future. This means that using contiguous data structures, such as arrays, help improve the execution times. This is because the contiguous block of memory will likely contain the other elements of the array, resulting in less **cache misses**.
+
+An example of a Data Structure that does not do well in this aspect is [Linked List](https://www.tutorialspoint.com/data_structures_algorithms/linked_list_algorithms.htm). In a Linked List, each node can be stored anywhere on the memory. This means that there will likely be more cache misses when trying to iterate through a Linked List. This can cause Linked Lists to be much slower than what you would expect in theory.
+
+#### Row/Column Major Order
+
+This rule is about how multidimensional arrays are stored in memory. Different programming languages have different methods of storing multidimensional arrays. [More Information](https://en.wikipedia.org/wiki/Row-_and_column-major_order).
+
+Using the incorrect method of access can cause many cache misses, resulting in a much slower execution time. Thus, it is important to be aware of which major order the programming language is using.
+
+For example, Java uses Row Major Order. We can create a test to see how big an impact using the wrong Major Order can be on the execution time.
+
+```
+int size = 10000;
+int[][] arr = new int[size][size];
+int x = 0;
+ 
+//Row Major order accessing
+long time = System.currentTimeMillis();
+for (int i = 0; i < size; i++) {
+    for (int j = 0; j < size; j++) {
+        x += arr[i][j];
+    }
+}
+System.out.println("Row major: " + (System.currentTimeMillis() - time));
+ 
+//Column Major order accessing
+time = System.currentTimeMillis();
+for (int i = 0; i < size; i++) {
+    for (int j = 0; j < size; j++) {
+        x += arr[j][i];
+    }
+}
+System.out.println("Column major: " +(System.currentTimeMillis() - time));
+```
+
+In the example above, Row major takes around 100ms, while column major takes around 2000ms. You can use the codes above and try it on your own too.
```

Comment:
>In the example above, Row major takes around 100ms, while column major takes around 2000ms.

These are the values you got by running the test in a typical notebook, right? Might want to mention that.